### PR TITLE
Converted SaveSnippetToS3 lambda from node.js to python

### PR DESCRIPTION
**Pull request by jdmaryville**
## Description

Converted the Save SnippetToS3 Lambda from node.js to python
## Motivation and Context

Increases readability and removes asynchronous functions

Fixes #195
## Checklist:

- [x] My code follows the code style of this project. 
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: lambdas/SaveSnippetToS3/index.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    authorizeTokenName = os.environ['authorizeTokenName']
+
+    #Invoke the auth. lambda to verify the accessToken mathches the userID for the requested resource
+    lambda_auth = client.invoke(
+        FunctionName=os.environ['authorizeTokenName'],
+        Payload=json.dumps({'accessToken': accessToken, 'userID': userID}))
+
+    lambda_payload_resp = json.loads(lambda_auth['Payload'].read())
+
+    if(lambda_payload_resp['statusCode'] == '400'):
```
Comment (dane-johnson) [inline lambdas/SaveSnippetToS3/index.py]: Parens around if statements are not necessary

Comment (thebho) [inline lambdas/SaveSnippetToS3/index.py]: Better safe than sorry

Comment (dane-johnson) [inline lambdas/SaveSnippetToS3/index.py]: If there is an error parens aren't going to help. As a python developer, I look at these parens and wonder if there is a function call or a tuple destructuring I am not seeing. Unnecessary and non-conventional

Comment (thebho) [inline lambdas/SaveSnippetToS3/index.py]: Oh, come on with your "As a python developer..."  Parens after an if mean the same in every language, even if you don't *have* to have them

Comment (dane-johnson) [inline lambdas/SaveSnippetToS3/index.py]: Python parens mean function calls, tuples, or order-of-operations overrides. This is none of those things.

Comment (thebho) [inline lambdas/SaveSnippetToS3/index.py]: It's totally an order of operations thing,  it converts `a == b` to a boolean and then uses that for conditional logic

Comment (dane-johnson) [inline lambdas/SaveSnippetToS3/index.py]: `a == b` is a boolean. You wouldn't write `c = ( a + b ) `

Comment (thebho) [inline lambdas/SaveSnippetToS3/index.py]: I might, it wouldn't be wrong

**Pull request closed without merging**